pkg/karmadactl/get.go

```diff
@@ -581,15 +605,25 @@ func clusterInfoInit(g *CommandGetOptions, karmadaConfig KarmadaConfig, clusterI
 			g.Clusters = append(g.Clusters, c)
 		}
 	}
-	return karmadaclient, nil
+	return karmadaRestConfig, nil
 }
 
 func getFactory(clusterName string, clusterInfos map[string]*ClusterInfo) cmdutil.Factory {
 	kubeConfigFlags := NewConfigFlags(true).WithDeprecatedPasswordFlag()
 	// Build member cluster kubeConfigFlags
-	kubeConfigFlags.BearerToken = stringptr(clusterInfos[clusterName].BearerToken)
 	kubeConfigFlags.APIServer = stringptr(clusterInfos[clusterName].APIEndpoint)
-	kubeConfigFlags.CaBundle = stringptr(clusterInfos[clusterName].CAData)
+
+	if clusterInfos[clusterName].KubeConfig != "" {
+		// Use kubeconfig to access member cluster
+		kubeConfigFlags.KubeConfig = stringptr(clusterInfos[clusterName].KubeConfig)
+		kubeConfigFlags.Context = stringptr(clusterInfos[clusterName].Context)
+		kubeConfigFlags.usePersistentConfig = true
+	} else {
+		// Use BearerToken to access member cluster
+		kubeConfigFlags.BearerToken = stringptr(clusterInfos[clusterName].BearerToken)
+		kubeConfigFlags.CaBundle = stringptr(clusterInfos[clusterName].CAData)
```

Comment:
Are we going to remove this logic at latter?

Reply:
Yes, because it is used by `promote` now, we can clean up it at latter.